intellij/src/saros/intellij/ui/views/buttons/AddContactButton.java

```diff
@@ -0,0 +1,167 @@
+    connectionHandler.addConnectionStateListener(connectionStateListener);
+
+    addActionListener(actionEvent -> showAddContactDialog());
+
+    if (!connectionHandler.isConnected()) {
+      setEnabled(false);
+    }
+  }
+
+  @Override
+  public void dispose() {
+    connectionHandler.removeConnectionStateListener(connectionStateListener);
+  }
+
+  /**
+   * Shows the dialogs requesting the JID and (optional) nickname for the account to add as a
+   * contact.
+   *
+   * <p>Calls {@link XMPPContactsService#addContact(JID, String, BiPredicate)} if the dialogs were
+   * not canceled.
+   */
+  private void showAddContactDialog() {
+
+    // TODO better communicate issue to user
+    // TODO use checkInput as it disables the OK button instead of just doing nothing when OK is
+    //  clicked on illegal input.
+    InputValidator inputValidator =
+        new InputValidator() {
+          @Override
+          public boolean checkInput(String inputString) {
+            return true;
+          }
+
+          @Override
+          public boolean canClose(String inputString) {
+            JID jid = new JID(inputString);
+
+            return jid.isValid() && !jid.getName().isEmpty();
+          }
+        };
+
+    String userID;
+
+    try {
+      userID =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_jid_dialog_message,
+              Messages.AddContactButton_contact_jid_dialog_initial_input,
+              Messages.AddContactButton_contact_jid_dialog_title,
+              inputValidator,
+              new TextRange(0, 0));
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
+
+      // TODO show error?
+      // showAccountCreationFailedError(e);
+
+      return;
+    }
+
+    if (userID == null) {
+      log.debug("Adding contact canceled by user during user id entry.");
+
+      return;
+    }
+
+    JID jid = new JID(userID);
+
+    String nickname;
+
+    try {
+      nickname =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_nickname_dialog_message,
+              "",
+              Messages.AddContactButton_contact_nickname_dialog_title);
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
+
+      // TODO show error?
+      // showAccountCreationFailedError(e);
+
+      return;
+    }
+
+    if (nickname == null) {
+      log.debug("Adding contact canceled by user during user nickname entry.");
+
+      return;
+    }
+
+    BiPredicate<String, String> dialog =
+        (title, message) -> {
+          log.fatal("showing dialog " + title + " - " + message);
+
+          Integer integer;
+
+          try {
+            integer = SafeDialogUtils.showYesNoDialog(project, message, title);
+          } catch (IllegalAWTContextException e) {
+            log.error("Tried to run dialog in an illegal context", e);
```

Comment:
Cannot happen here as it would already happen on line 123.

Reply:
Yeah, but I need to catch it anyway. And I will most likely handle the user reporting in a separate method, so it shouldn't add much code to handle this case the same as the first call.